postgres/datadog_checks/postgres/statement_samples.py

```diff
@@ -359,28 +364,30 @@ def _run_explain(self, dbname, statement, obfuscated_statement):
             return result[0][0]
 
     def _run_explain_safe(self, dbname, statement, obfuscated_statement):
-        if not self._can_explain_statement(obfuscated_statement):
-            return None
+        # type: (str, str, str) -> Tuple[Union[Dict, None], Union[str, None], Union[str, None]]
+        ok, db_explain_error = self._can_explain_statement(obfuscated_statement)
+        if not ok:
+            return None, db_explain_error.value, None
 
-        explain_setup_state = self._get_db_explain_setup_state_cached(dbname)
-        if explain_setup_state != DBExplainSetupState.ok:
+        db_explain_error, err = self._get_db_explain_setup_state_cached(dbname)
+        if db_explain_error is not None:
             self._check.count(
                 "dd.postgres.statement_samples.error",
                 1,
-                tags=self._dbtags(dbname, "error:explain-{}".format(explain_setup_state)),
+                tags=self._dbtags(dbname, "error:explain-{}".format(db_explain_error)),
             )
-            return None
+            return None, db_explain_error.value, '{}'.format(type(err))
```

Comment:
It would be clearer to return the error as-is and let the caller format it closer to where it's going to be used.

Reply:
Also true of the error. Return the enum and use the `.value` when constructing the event.